filters/auth/oidc.go

```diff
@@ -186,6 +188,21 @@ func (s *tokenOidcSpec) CreateFilter(args []interface{}) (filters.Filter, error)
 		}
 	}
 	log.Debugf("Auth Code Options: %v", f.authCodeOptions)
+
+	// inject additional headers from the access token for upstream applications
+	if len(sargs) > 7 && sargs[7] != "" {
+		f.upstreamHeaders = make(map[string]string)
+
+		for _, header := range strings.Split(sargs[7], " ") {
+			sl := strings.SplitN(header, ":", 2)
+			if len(sl) != 2 || sl[0] == "" || sl[1] == "" {
+				return nil, fmt.Errorf("%v: malformatted filter for upstream headers %s", filters.ErrInvalidFilterParameters, sl)
```

Comment:
Do you know if you do %w instead of %v, the error.Is() function can check if it’s a ErrInvalidFilterParameters?
I always wanted to move the errors in skipper from %v to %w and maybe you already have experience with it.

Reply:
Great point! Actually I also work on migrating all `fmt.Errorf ...%v to %w` syntax, it is cost-free.
`error.Is()` is one new possibility, but actually the wrapping includes the underlying error type which can be retrieve via normal switch case type statements again. Cannot promote its use case enough as it allows way more safer error tests.